purchases/src/main/kotlin/com/revenuecat/purchases/paywalls/components/PaywallAnimation.kt

```diff
@@ -0,0 +1,64 @@
+package com.revenuecat.purchases.paywalls.components
+
+import com.revenuecat.purchases.InternalRevenueCatAPI
+import com.revenuecat.purchases.utils.serializers.SealedDeserializerWithDefault
+import dev.drewhamilton.poko.Poko
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+/**
+ * Defines an animation to be used for paywall transitions.
+ *
+ * @property type The type of animation to use like ease in, ease out, etc.
+ * @property msDelay The delay in milliseconds before the animation starts.
+ * @property msDuration The duration in milliseconds of the animation.
+ */
+@Suppress("unused")
+@InternalRevenueCatAPI
+@Poko
+@Serializable
+@SerialName("animation")
+class PaywallAnimation(
+    @get:JvmSynthetic val type: AnimationType,
+    @get:JvmSynthetic
+    @SerialName("ms_delay")
+    val msDelay: Int?,
+    @get:JvmSynthetic
+    @SerialName("ms_duration")
+    val msDuration: Int?,
+) {
+
+    /**
+     * Defines the type of animation to use for paywall transitions.
+     *
+     * [NOTE] This is a sealed class and not an enum because we see a future where we may want
+     * to pass back more verbose instructions to the view layer than a simple enum case
+     */
+    @InternalRevenueCatAPI
```

Comment:
NABD, but inner classes/enums of an `@InternalRevenueCatAPI` do not need to be annotated again so we could remove this annotation from these.